### PR TITLE
chore: node-core cleanup

### LINKED ISSUE #6176: Node-core cleanup pass
### Describe the feature

after extracting some parts of the reth cli crate into node-core, we should do a cleanup pass

1. Move NodeConfig to core
2. get rid of duplicate vergen build.rs (ideally remove it from core)
3. Move utils.rs from core back to cli crate
...

 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Rjected**
Fixes https://github.com/paradigmxyz/reth/issues/6176

This does a few main things:
* moves `NodeConfig` to `node-core`, along with any required methods or helper structs.
* moves `DbTool` back to `bin/reth`. This was the only thing in `utils` that was not shared by the `NodeConfig`
* remove vergen dependency from `bin/reth`. It is needed in `node-core` because network args, for example, includes vergen-generated info in the default client version.
* moves `EngineApiStore` to `node-core`

re-exports are added to prevent breakage

### CODE REVIEW

File: bin/reth/build.rs
Comment (mattsse) [inline bin/reth/build.rs:1]: hmm, I'd rather keep this and delete the other one if possible

**Review by mattsse — APPROVED** [review]
this is better, though build broken

**Comment by Rjected** [comment]
fixing lints

File: bin/reth/build.rs
Comment (Rjected) [reply bin/reth/build.rs:1]: so `NetworkArgs` has this:
https://github.com/paradigmxyz/reth/blob/bfa951d8ae27670774fcb3f3182a01a07103e83d/crates/node-core/src/args/network_args.rs#L42-L44

which relies on vergen
https://github.com/paradigmxyz/reth/blob/bfa951d8ae27670774fcb3f3182a01a07103e83d/crates/node-core/src/version.rs#L59-L66

If we keep the `build.rs` in `bin/reth` will this still work?

Comment (Rjected) [reply bin/reth/build.rs:1]: tried it, does not work, rip

**Pull request merged**